### PR TITLE
Sub reports for Aggregate Report

### PR DESCRIPTION
**Pull request by Rieven**
### Changes
The initial idea was to show also sub reports for Aggregate Report. This was missing, it was only available in bytes but we can see al sub reports which contributed to the Aggregation of these sub reports into the Aggregate Report.

### Issue link
https://github.com/minvws/nl-kat-coordination/issues/3805

Closes https://github.com/minvws/nl-kat-coordination/issues/3805

### Demo

image

All sub reports for Aggregate Report:
image

### QA notes

_Please add some information for QA on how to test the newly created code._

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [x] Tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by ammar92 — COMMENTED** [review]
Tiny suggestion, looks good otherwise

File: rocky/reports/views/mixins.py [hunk 1]
```diff
@@ -208,8 +211,7 @@ def save_aggregate_report_data(
     report_recipe: Reference | None = None,
 ) -> Report:
     observed_at = get_observed_at
-
-    now = datetime.utcnow()
```
Comment (Rieven) [inline rocky/reports/views/mixins.py:212]: deprecated

File: rocky/reports/report_types/definitions.py [hunk 2]
```diff
@@ -168,6 +168,11 @@ class AggregateReportSubReports(TypedDict):
     optional: list[type[Report]]
 
 
+class SubReportPlugins(TypedDict):
+    required: list[type[Report]]
+    optional: list[type[Report]]
```
Comment (ammar92) [inline rocky/reports/report_types/definitions.py:173]: Why is this one added? It's a duplicate of the above type and it's not even used. I suggest removing it or rename the `AggregateReportSubReports` type to a more generic name

File: rocky/reports/report_types/definitions.py [hunk 3]
```diff
@@ -17,6 +17,11 @@ class ReportPlugins(TypedDict):
     optional: set[str]
 
 
+class SubReportPlugins(TypedDict):
+    required: list[str]
+    optional: list[str]
```
Comment (Rieven) [inline rocky/reports/report_types/definitions.py:22]: @ammar92 I have realized that the TypeDict of:
image

is a different one. It is a dict of type Report, buy this is another one. I was confused too. 
It is this one which looks like the plugins:
image

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [ ] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Aggregate reports are available again! 

### What doesn't work:
See below. 

### Bug or feature?:
Creating an Aggregate report for hostname mispo.es with all reports selected results in only the IPv6 report being visible. Somehow the Vulnerabilities report is shown in the ToC but not in the report itself. All subreports are present and can be opened.  

image

**Review by underdarknl — APPROVED** [review]

**Pull request merged**